### PR TITLE
file upload updated

### PR DESCRIPTION
**Pull request by Michaeljogoh**
added more features to the project

### CODE REVIEW

File: controller/fileController.js [hunk 1]
```diff
@@ -1,6 +1,63 @@
...
+
+
+const checkFileType = (file, cb) => {
+    //Allow extensions
+    const filetypes = /jpeg|jpg|png|gif|JPG/;
+    //check ext
+    const extname = filetypes.test(path.extname 
+    (file.originalname).toLowerCase());
+    //check mine
+    const mimetype = filetypes.test(file.mimetype);
```
Comment (Asylcreek) [inline controller/fileController.js:34]: You can do `file.mimetype.startsWith('image')`

File: controller/fileController.js [hunk 2]
```diff
@@ -1,6 +1,63 @@
...
+        
+    } else {
+        const newFile =  new fileModel({
+        title:req.body.title,
+        image: {
+            data:req.body.filename,
+            contentType:"image"
+        }
+            }); 
+                newFile.save();
```
Comment (Asylcreek) [inline controller/fileController.js:55]: `await newFile.save()`

File: controller/fileController.js [hunk 3]
```diff
@@ -1,6 +1,63 @@
...
+    const mimetype = filetypes.test(file.mimetype);
+
+    if(mimetype && extname){
+        return cb(null, true);
+    }else{
+        cb('Error: Images only');
+    }
+}
+
+upload(req , res, (err)=>{
```
Comment (Asylcreek) [inline controller/fileController.js:43]: 1.There's no function keyword
2. Express passes `next` as the third argument, not `error`
3. You are missing one ')'

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: controller/fileController.js (hunk 1 above)
Comment (Michaeljogoh) [reply controller/fileController.js:34]: ok i will change that

File: controller/fileController.js (hunk 2 above)
Comment (Michaeljogoh) [reply controller/fileController.js:55]: ok

File: controller/fileController.js (hunk 3 above)
Comment (Michaeljogoh) [reply controller/fileController.js:43]: ok

File: controller/fileController.js [hunk 4]
```diff
@@ -39,25 +37,43 @@ const checkFileType = (file, cb) => {
         cb('Error: Images only');
     }
 }
-
-upload(req , res, (err)=>{
-    if(err){
-        console.log(err);
-        
-    } else {
+upload  (req , res, next =>{
```
Comment (Asylcreek) [inline controller/fileController.js:40]: `const  upload = (req, res, next) => {`
OR
`function upload (req, res, next) {`

File: controller/fileController.js [hunk 5]
```diff
@@ -39,25 +37,43 @@ const checkFileType = (file, cb) => {
...
         }
-            }); 
-                newFile.save();
+            });
+             
+        newFile.save();
                 res.status(200);
-                res.json("File Uploaded");
-            }
+                res.json({newFile});
```
Comment (Asylcreek) [inline controller/fileController.js:51]: `res.status(200).json(newFile)`

**Review by Asylcreek — CHANGES_REQUESTED** [review]

**Pull request merged**